File: cpl/src/main/java/com/enrico/colorpicker/CircleDrawable.java
package com.enrico.colorpicker;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.graphics.PixelFormat;
import android.graphics.Rect;
import android.graphics.drawable.Drawable;
import android.support.annotation.ColorInt;
import android.support.annotation.FloatRange;
import android.support.annotation.IntRange;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

@SuppressWarnings("WeakerAccess")
public class CircleDrawable extends Drawable {

    private final Paint mInnerPaint;
    private final Paint mOuterPaint;
    private int mRadius = 0;

    public CircleDrawable(final int color) {
        mInnerPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        mInnerPaint.setColor(color);
        mOuterPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        mOuterPaint.setColor(shiftColorDown(color));
    }

    @ColorInt
    public static int shiftColor(@ColorInt int color,
                                 @FloatRange(from = 0.0f, to = 2.0f) float by) {
        if (by == 1f) {
            return color;
        }
        float[] hsv = new float[3];
        Color.colorToHSV(color, hsv);
        hsv[2] *= by; // value component
        return Color.HSVToColor(hsv);
    }

    @ColorInt
    public static int shiftColorDown(@ColorInt int color) {
        return shiftColor(color, 0.9f);
    }

    @Override
    public void draw(@NonNull Canvas canvas) {
        final Rect bounds = getBounds();
        canvas.drawCircle(bounds.centerX(), bounds.centerY(), mRadius, mOuterPaint);
        canvas.drawCircle(bounds.centerX(), bounds.centerY(), mRadius - 5, mInnerPaint);
    }

    @Override
    protected void onBoundsChange(final Rect bounds) {
        super.onBoundsChange(bounds);
        mRadius = Math.min(bounds.width(), bounds.height()) / 2;
    }

    @Override
    public void setAlpha(@IntRange(from = 0, to = 255) int alpha) {
        mInnerPaint.setAlpha(alpha);
        mOuterPaint.setAlpha(alpha);
    }

    public void setColor(int color) {
        if (color != mInnerPaint.getColor()) {
            mInnerPaint.setColor(color);
            mOuterPaint.setColor(shiftColorDown(color));
            invalidateSelf();
        }
    }

    @Override
    public void setColorFilter(@Nullable ColorFilter colorFilter) {
    }

    @Override
    public int getOpacity() {
        return PixelFormat.TRANSLUCENT;
    }
}

File: app/src/main/java/com/brit/swiftinstaller/ui/CircleDrawable.kt
package com.brit.swiftinstaller.ui

import android.graphics.*
import android.graphics.drawable.Drawable
import androidx.annotation.ColorInt
import androidx.annotation.FloatRange
import androidx.annotation.IntRange

class CircleDrawable(color: Int) : Drawable() {

    private val mInnerPaint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
    private val mOuterPaint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
    private var mRadius = 0

    init {
        mInnerPaint.color = color
        mOuterPaint.color = shiftColorDown(color)
    }

    override fun draw(canvas: Canvas) {
        val bounds = bounds
        canvas.drawCircle(bounds.centerX().toFloat(), bounds.centerY().toFloat(), mRadius.toFloat(), mOuterPaint)
        canvas.drawCircle(bounds.centerX().toFloat(), bounds.centerY().toFloat(), (mRadius - 5).toFloat(), mInnerPaint)
    }

    override fun onBoundsChange(bounds: Rect) {
        super.onBoundsChange(bounds)
        mRadius = Math.min(bounds.width(), bounds.height()) / 2
    }

    override fun setAlpha(@IntRange(from = 0, to = 255) alpha: Int) {
        mInnerPaint.alpha = alpha
        mOuterPaint.alpha = alpha
    }

    @Suppress("unused")
    fun setColor(color: Int) {
        if (color != mInnerPaint.color) {
            mInnerPaint.color = color
            mOuterPaint.color = shiftColorDown(color)
            invalidateSelf()
        }
    }

    override fun setColorFilter(colorFilter: ColorFilter?) {}

    override fun getOpacity(): Int {
        return PixelFormat.TRANSLUCENT
    }

    companion object {

        @Suppress("MemberVisibilityCanBePrivate")
        @ColorInt
        fun shiftColor(@ColorInt color: Int,
                       @FloatRange(from = 0.0, to = 2.0) by: Float): Int {
            if (by == 1f) {
                return color
            }
            val hsv = FloatArray(3)
            Color.colorToHSV(color, hsv)
            hsv[2] *= by // value component
            return Color.HSVToColor(hsv)
        }

        @ColorInt
        fun shiftColorDown(@ColorInt color: Int): Int {
            return shiftColor(color, 0.9f)
        }
    }
}
